src/types/GlyphWidth.cpp

```diff
@@ -35,9 +35,9 @@ bool IsGlyphFullWidth(const wchar_t wch) noexcept
 // - pfnFallback - the function to use as the fallback method.
 // Return Value:
 // - <none>
-void SetGlyphWidthFallback(std::function<bool(const std::wstring_view)> pfnFallback)
+void SetGlyphWidthFallback(std::function<bool(const std::wstring_view&)> pfnFallback) noexcept

```

Comment:
i thought taking string views by reference was more expensive than by copy (because of the ABI or whatever)

Reply:
Now that VS 17.3 includes a fix which makes passing 128-bit structs much faster on x64 in many situations, this is a huge "it depends" now. But before that, passing by reference was almost always faster. This is because the MSVC compiler often copied those parameters onto the stack with SSE instructions even if it didn't need to and 1-2 regular `mov`s would've been sufficient.

Reply:
Maybe my example is wrong, but looks like same instructions would generated by MSVC https://godbolt.org/z/W9P9WboEq

Reply:
I joust found way to reproduce this old behaviour https://godbolt.org/z/vqqWjj8d8

Reply:
Yep. The issue used to occur if string-views are passed by-value by just a single level of indirection: https://godbolt.org/z/vh8c3sa17

This issue is especially glaring when used with the STL where all string-views are passed by-value externally and internally and you couldn't avoid this issue: https://godbolt.org/z/cPra48qqY
I'm not sure if our CI is on MSVC 19.33 already but if it is I think we can safely stop passing string-views by-reference. 🙂

Reply:
Thanks for information